base/string.jl

```diff
@@ -1,11 +1,21 @@
 ## core text I/O ##
 
 print(io::IO, x) = show(io, x)
-function print(io::IO, xs...)
-    lock(io) do io
-        for x in xs print(io, x) end
+stagedfunction print(io::IO, xs...)
+    if :lock in fieldnames(io)
```

Comment:
I never realised that! `isdefined` adds around 1.5 - 2.0 seconds to the results.